src/ApplicationInsights.Kubernetes/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -18,14 +18,16 @@ public static partial class ApplicationInsightsExtensions
         /// Enables Application Insights for Kubernetes on the Default TelemetryConfiguration in the dependency injection container with custom options.
         /// </summary>
         /// <param name="services">Collection of service descriptors.</param>
-        /// <param name="applyOptions">Action to customize the configuration of Application Insights for Kubernetes.</param>
         /// <param name="diagnosticLogLevel">Sets the diagnostics log levels for the enricher.</param>
+        /// <param name="skipRegisterBackendService">Allows skipping register any backend service for constrained environment like InProc Azure Function.</param>
```

Comment:
The fact that the "registration" for backend services is "skipped" is an implementation detail. Consider `disableBackgroundServices` as a name for this parameter.

The description could then say something along the lines of the following (emphasizing user impact):

> If true, the Application Insights enricher library will not use any background (hosted) services,, and Kubernetes information will not be fetched automatically. Hosted services are not allowed in some environments, e.g. Azure Function. For more information see https://relevant/wiki/page.md

Reply:
Agree on the naming of the parameter.

I understand the advantages of the auto-detect azure function, here are the considerations:

* It is still not fully automatic at this point - the user will still need to write some other code to bootstrap the detector on their own. And probably a document is still needed.

I say we do it in 2 stages - use this to unblock the scenario. Use another PR in the future to fully automate the scenario - I still can't find a good place to fun background service in Azure Function at this moment.

What do you think?

Reply:
Renamed the parameter name